runtime/Python3/src/antlr4/TokenStreamRewriter.py

```diff
@@ -165,28 +175,31 @@ def _reduceToSingleOperationPerIndex(self, rewrites):
         for i, iop in enumerate(rewrites):
             if any((iop is None, not isinstance(iop, TokenStreamRewriter.InsertBeforeOp))):
                 continue
-            prevInserts = [op for op in rewrites[:i] if isinstance(iop, TokenStreamRewriter.InsertBeforeOp)]
-            for prevIop in prevInserts:
-                if prevIop.index == iop.index:
+            prevInserts = [op for op in rewrites[:i] if isinstance(op, TokenStreamRewriter.InsertBeforeOp)]
+            for i, prevIop in enumerate(prevInserts):
```

Comment:
are you sure about reusing i in an inner loop?

Reply:
Yeah, I have noticed this tricky code. The same code appears in python-2-runtime so I just copied it.

In the inner loop, only ``i`` of ``enumerate(prevInserts)`` is used. And when the inner loop ends, ``i`` becomes ``len(prevInserts) - 1`` which depends on the operation sequence.

The core problem is that ``for`` loop is not a scope in python, so the ``i`` of the inner-loop just covers the outter ``i``.

Since you point this out, I can fix the bug in py2 & py3.